lib/mix/test/mix/tasks/compile.all_test.exs

```diff
@@ -0,0 +1,28 @@
+Code.require_file "../../test_helper.exs", __DIR__
+
+defmodule Mix.Tasks.CompileAllTest do
```

Comment:
I don't think we need this test. compile.all is an implementation detail, it is more important to check that the "compile" task returns what you want.

Reply:
To be more precise, I would move this test to mix/tasks/compiler_test.exs.

Reply:
Oh, I see. You want to invoke `compile.all` from your editor since `compile` exits. I don't think we should expose `compile.all` at all. If you would like for the `compile` task to not exit on error, we can provide a flag that does that. `--return-errors` or something of sorts.

Reply:
Sounds good! I'll make that change.